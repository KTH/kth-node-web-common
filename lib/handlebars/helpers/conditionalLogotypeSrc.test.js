const Handlebars = require('handlebars')
const { conditionalLogotypeSrc, registerConditionalLogotypeSrcHelper } = require('./conditionalLogotypeSrc')

jest.mock('handlebars')

Handlebars.SafeString = jest.fn().mockImplementation(str => ({ string: str }))

describe('registerConditionalLogotypeSrcHelper', () => {
  it('Registers conditionalLogotypeSrc helper', () => {
    registerConditionalLogotypeSrcHelper()
    expect(Handlebars.registerHelper).toHaveBeenCalledWith('conditionalLogotypeSrc', conditionalLogotypeSrc)
  })
})

describe('conditionalLogotypeSrc', () => {
  it('should throw an error if proxyPrefix is not a string', () => {
    const theme = 'external'
    const invalidProxyPrefix = 123

    expect(() => {
      conditionalLogotypeSrc(theme, invalidProxyPrefix)
    }).toThrow(new Error('Invalid proxyPrefix: must be a string'))
  })

  it('should return logotype-white.svg for external theme', () => {
    const theme = 'external'
    const proxyPrefix = '/prefix'

    const result = conditionalLogotypeSrc(theme, proxyPrefix)
    expect(result.string).toMatchSnapshot()
  })

  it('should return logotype-blue.svg for intranet theme', () => {
    const theme = 'intranet'
    const proxyPrefix = '/prefix'

    const result = conditionalLogotypeSrc(theme, proxyPrefix)
    expect(result.string).toMatchSnapshot()
  })

  it('should return logotype-blue.svg for student-web theme', () => {
    const theme = 'student-web'
    const proxyPrefix = '/prefix'

    const result = conditionalLogotypeSrc(theme, proxyPrefix)
    expect(result.string).toMatchSnapshot()
  })
})
