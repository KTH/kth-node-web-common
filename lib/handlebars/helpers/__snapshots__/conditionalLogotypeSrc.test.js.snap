// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conditionalLogotypeSrc should return logotype-blue.svg for intranet theme 1`] = `"/prefix/assets/logotype/logotype-blue.svg"`;

exports[`conditionalLogotypeSrc should return logotype-blue.svg for student-web theme 1`] = `"/prefix/assets/logotype/logotype-blue.svg"`;

exports[`conditionalLogotypeSrc should return logotype-white.svg for external theme 1`] = `"/prefix/assets/logotype/logotype-white.svg"`;
